### ISSUE TITLE
Sometimes DepCells don't re-execute its function when dependencies change

### ISSUE THREAD

**Issue by chungwu** [bobtail-dev/bobtail-core#13]
Reproduced thusly (bobtail-rx version 2.1.3):

```javascript
stateCell = rx.cell({state: "unloaded"})
loadingCell = rx.bind(function() {return stateCell.get().state == "loading";})
orgsCell = rx.bind(function() {return stateCell.get().data;})
filteredOrgsCell = rx.bind(function() {
    orgs = orgsCell.get()
    if (orgs && orgs.length > 0) {
        return orgs.filter(function(o) { return o.name.indexOf("Acme") >= 0;});
    } else {
        return orgs;
    }
})
body = rx.bind(function() {
    if (loadingCell.get()) {
        console.log("loading");
        return false;
    }
    console.log("filteredOrgs", filteredOrgsCell.get());
    return filteredOrgsCell.get();
})

// this prints "loading"
stateCell.set({state: "loading"})

// this prints "filteredOrgs undefined", though filteredOrgs should be defined
stateCell.set({state: "loaded", data: [{name:"Acme"}, {name: "Infer"}]})

// you can firm that filteredOrgsCell does have the right thing
filteredOrgCell.get()

// doing this again will print "filteredOrgs [{...}]" (the correct answer)
stateCell.set({state: "loaded", data: [{name:"Acme"}, {name: "Infer"}]})
```

One random guess: It looks like when `stateCell.set({state: "loaded", data: [...]})` is called, it propagated the change to the `body` cell first, before the `filteredOrgsCell`.  But once the change is propagated to `filteredOrgsCell`, the `body` cell is not recomputed again, even though it also depends on `filteredOrgsCell`.

**Comment by rmehlinger** [comment]
Well, that's a problem. When we refresh a cell, we compute all of its downstream dependencies ahead of time in topological order. We 'shield' each cell, so that it is not refreshed multiple times. However, in this case, when we set `stateCell` to `'loading'`,  `filteredOrgsCell` removes `body` as a dependency. When we set `stateCell` again, we resubscribe `filteredOrgsCell` as an upstream dependency of `body`, but crucially we did not know that when we began refreshing. That is why the dependencies are refreshed out of order, and the shielding mechanism prevents `body` from being refreshed a second time.

Fixing this will be tricky, but shouldn't be impossible.
